src/svg/qsvggenerator.cpp

```diff
@@ -997,8 +1007,37 @@ void QSvgPaintEngine::updateState(const QPaintEngineState &state)
     // close old state and start a new one...
     if (d->afterFirstUpdate)
         *d->stream << "</g>\n\n";
+    
+    QPainter* p = painter();
+    if (p->hasClipping()) {
+        QPainterPath painter_path = p->clipPathF();
+
+        if (painter_path.elementCount() > 0) {
+            QString clip_path = createPath(painter_path);
+
+            bool path_does_not_exist = 0 == clip_path_to_id.count(clip_path);
+            if (path_does_not_exist) {
+                clip_path_to_id[clip_path] = clip_counter++;
+                
+                d->stream->setString(&d->defs);
+
+                *d->stream << "<clipPath id=\"clip" << clip_path_to_id[clip_path] << "\">" << '\n';
```

Comment:
I think if you changed:
`clip_path_to_id[clip_path] = clip_counter++;` 
to
`clip_path_to_id[clip_path] = ++clip_counter;`
then `clip_counter` would be storing the clip ID of the _current_ clip, so you could then use it in the stream, rather than having to do the lookup again repeatedly.

Reply:
This is true only for the dictionary call inside the `if`, the one outside will need to grab it from the mapping if the next time this function is called and the same path already exists, it can't grab it from clip_counter.

Reply:
Yeah that's true, if the clipping changes then changes back to something already in the cache, you have to look it up again. Good point. Doesn't really save enough to be worth changing then.